addon/v1alpha1/types_clustermanagementaddon.go

```diff
@@ -175,10 +185,145 @@ type PlacementStrategy struct {
+	// The maximum concurrently updating number of addons.
+	// Value can be an absolute number (ex: 5) or a percentage of desired addons (ex: 10%).
+	// Absolute number is calculated from percentage by rounding up.
+	// Defaults to 25%.
+	// Example: when this is set to 30%, once the addon configs change, the addon on 30% of the selected clusters
+	// will adopt the new configs. When the addons with new configs are healthy, the addon on the remaining clusters
+	// will be further updated.
+	// +kubebuilder:default:="25%"
+	// +optional
+	MaxConcurrency *intstr.IntOrString `json:"maxConcurrency,omitempty"`
+}
+
+// RollingUpdateWithCanary represents the canary placement and behavior to rolling update add-on configurations
+// on the selected clusters.
+type RollingUpdateWithCanary struct {
+	// Canary placement reference.
+	// +kubebuilder:validation:Required
+	// +required
+	Placement PlacementRef `json:"placement,omitempty"`
+
+	// the behavior to rolling update add-on configurations.
+	RollingUpdate `json:",inline"`
 }
 
 // ClusterManagementAddOnStatus represents the current status of cluster management add-on.
 type ClusterManagementAddOnStatus struct {
+	// defaultconfigReferences is a list of current add-on default configuration references.
+	// +optional
+	DefaultConfigReferences []DefaultConfigReference `json:"defaultconfigReferences,omitempty"`
+	// installProgression is a list of current add-on configuration references per placement.
+	// +optional
+	InstallProgression []InstallProgression `json:"installProgression,omitempty"`
+}
+
+type InstallProgression struct {
+	PlacementRef `json:",inline"`
+
+	// configReferences is a list of current add-on configuration references.
+	// +optional
+	ConfigReferences []InstallConfigReference `json:"configReferences,omitempty"`
+
+	// conditions describe the state of the managed and monitored components for the operator.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"  patchStrategy:"merge" patchMergeKey:"type"`
+}
+
+// DefaultConfigReference is a reference to the current add-on configuration.
+// This resource is used to record the configuration resource for the current add-on.
+type DefaultConfigReference struct {
+	// This field is synced from ClusterManagementAddOn Configurations.
+	ConfigGroupResource `json:",inline"`
+
+	// desiredConfigSpecHash record the desired config spec hash.
+	DesiredConfigSpecHash *ConfigSpecHash `json:"desiredConfigSpecHash"`
+}
+
+// InstallConfigReference is a reference to the current add-on configuration.
+// This resource is used to record the configuration resource for the current add-on.
+type InstallConfigReference struct {
+	// This field is synced from ClusterManagementAddOn Configurations.
+	ConfigGroupResource `json:",inline"`
+
+	// desiredConfigSpecHash record the desired config spec hash.
+	DesiredConfigSpecHash *ConfigSpecHash `json:"desiredConfigSpecHash"`
```

Comment:
it seems to me the name should be desiredConfig rather than desiredConfigSpecHash, and in the desiredConfig, we have specHash/name/namespace